graphdatascience/arrow_client/v1/gds_arrow_client.py

```diff
@@ -0,0 +1,626 @@
+from __future__ import annotations
+
+import json
+import logging
+import re
+from types import TracebackType
+from typing import Any, Callable, Iterable, Type
+
+import pandas
+import pyarrow
+from neo4j.exceptions import ClientError
+from pyarrow import Array, ChunkedArray, DictionaryArray, RecordBatch, Table, chunked_array, flight
+from pyarrow.types import is_dictionary
+from pydantic import BaseModel
+
+from graphdatascience.arrow_client.arrow_endpoint_version import ArrowEndpointVersion
+from graphdatascience.arrow_client.authenticated_flight_client import AuthenticatedArrowClient
+from graphdatascience.arrow_client.v1.data_mapper_utils import deserialize_single
+
+from ...semantic_version.semantic_version import SemanticVersion
+
+
+class GdsArrowClient:
```

Comment:
i think we should introduce a GdsArrowClient user-facing interface which the different versioned clients can implement.
WDYT?